### DISCUSSION TITLE
Build script

### DISCUSSION THREAD

**Discussion by therion23** [jcorporation/myMPD#927] [Q&A] [answered]
10.1 and onwards (i think that is where i started).

Good day, Jürgen - i am having some trouble with the changes to the build script.

DESTDIR="/usr/local" ./build.sh

Before, i had a local installation in /usr/local/bin - now it puts mympd in /usr/local/usr/bin and gives me error messages about /var not being writable.

Notice, i run it as local user with /usr/local writable the old Debian way, but it always used to work, so - what am i doing wrong?

Thank you in advance, of course!

Bo

**Comment by jcorporation** [comment, marked as answer]
Sure you did it in the past with `DESTDIR`? You must set `CMAKE_INSTALL_PREFIX` to `/usr/local`.
```
cmake -B release -DCMAKE_INSTALL_PREFIX:PATH=/usr/local -DCMAKE_BUILD_TYPE=Release .
make -C release
make -C release install
```

Edit: found a bug in the workdir/cachedir calculation (fixed in devel).

**Comment by therion23** [comment]
Hello! No, i am in fact sure i used INSTALL_PREFIX in the past, and changed because it is no longer in the helptext :)

This looks a bit different from before 10.1 but has the same effect - however, i still get:

14:00:21 ERROR    mympd     Config dir: creating "/var/config" failed
14:00:21 ERROR    mympd     Permission denied

And if i try to move the data dir from /var/lib/mympd to /var/config, mympd cannot see its old data.

However, using your recipe and adding -a and -w to the startup script, that combination did the job. Never needed those before, but now it works like it used to - thank you so much!

Bo

**Reply by jcorporation** [reply]
> 14:00:21 ERROR mympd Config dir: creating "/var/config" failed

This is the mentioned bug. This will be solved with the 10.1.6 version.

**Comment by therion23** [comment]
Absolutely lovely. Thank you once again for your help!
